### DISCUSSION TITLE
How to define temperature-related properties in SolidMaterialProperties?

### DISCUSSION THREAD

**Discussion by Ethan-xj** [idaholab/moose#26543] [Q&A Modules: Thermal Hydraulics] [answered]
@joshuahansel Hi Josh. As the description in the title, should I use `ParsedFunction` to define temperature-related properties in SolidMaterialProperties? Are there any cases which I can refer to?

**Comment by Ethan-xj** [comment]
I understand now. In the ParsedFunction, 't' is used to represent temperature.

**Reply by GiudGiud** [reply]
`t` is used to represent time in functions in general
But when the function is used, temperature is passed as time

**Reply by Ethan-xj** [reply]
@GiudGiud Thank you. And I have another question.
How can I output the temperature of the heatstructure?
I set the parameter like this:
```
  [ROD21]
    type = HeatStructureCylindrical
    position = '0 21 0'
    orientation = '0 0 1'
    length = 0.6
    n_elems = 60
    initial_T = ${T_in}
    names = 'clad gap fuel air'
    widths = '0.0006 0.00015 0.0035 0.001'
    n_part_elems = '3 2 5 1'
    materials = 'clad-mat gap-mat fuel-mat air-mat'
  [../]
```
I want to get the temperature of these four parts: clad gap fuel and air. What should I do?

**Reply by GiudGiud** [reply]
Hello

I think you can use the PointValue postprocessor for this. 

I don't know if there is a dedicated heat structure variable output. @joshuahansel 

Guillaume

**Reply by joshuahansel** [reply, marked as answer]
Which do you want?

- temperature at a certain point in each: Guillaume answered that.
- average temperature in each region: use `AverageNodalVariableValue` with `variable = T_solid` and `block = R0D21:clad` for example.
- temperature at every point: just output to Exodus

**Reply by Ethan-xj** [reply]
Thank you very much !
